examples/conditionals.yaml

```diff
@@ -1,35 +1,41 @@
 # Conditionals provide a way to affect the control flow of a
 # workflow at runtime, depending on parameters. In this example
-# the 'print-hello' template may or may not be executed depending
-# on the input parameter, 'should-print'. When submitted with:
+# the 'printHello' template may or may not be executed depending
+# on the input parameter, 'shouldPrint'. When submitted with:
 #   argo submit examples/conditionals.yaml
-# the step will be skipped since 'should-print' will evaluate false.
+# the step will be skipped since 'shouldPrint' will evaluate false.
 # When submitted with:
-#   argo submit examples/conditionals.yaml -p should-print=true
-# the step will be executed since 'should-print' will evaluate true.
+#   argo submit examples/conditionals.yaml -p shouldPrint=true
+# the step will be executed since 'shouldPrint' will evaluate true.
 apiVersion: argoproj.io/v1alpha1
 kind: Workflow
 metadata:
   generateName: conditional-
 spec:
-  entrypoint: conditional-example
+  entrypoint: conditionalExample
   arguments:
     parameters:
-    - name: should-print
-      value: "false"
+    - name: shouldPrint
+      value: "true"
 
   templates:
-  - name: conditional-example
+  - name: conditionalExample
     inputs:
       parameters:
-      - name: should-print
+      - name: shouldPrint
     steps:
-    - - name: print-hello
-        template: whalesay
-        when: "{{inputs.parameters.should-print}} == true"
+    - - name: printHello-govaluate
+        template: argosay
+        when: "{{inputs.parameters.shouldPrint}} == true" # govaluate form
+      - name: printHello-expr
+        template: argosay
+        when: "{{= inputs.parameters.shouldPrint == 'true'}}" # expr form
+      - name: printHello-expr-json
+        template: argosay
+        when: "{{=jsonpath(workflow.parameters.json, '$[0].value') == 'true'}}" # expr form
 
-  - name: whalesay
+  - name: argosay
     container:
-      image: docker/whalesay:latest
+      image: argoproj/argosay:v1
       command: [sh, -c]
```

Comment:
had to change the whalesay image to argosay image since apparently whalesay isn't supposed to be used for e2e tests